.github/workflows/nix_ci.yml

```diff
@@ -31,6 +31,95 @@ jobs:
         signingKey: '${{ secrets.CACHIX_SIGNING_KEY }}'
         # If you chose API tokens for write access OR if you have a private cache
         authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
-    - run: nix build .#game-night-backend
-    - run: nix build .#game-night-frontend
-    - run: nix build .#game-night-docker
+    
+    - name: Build Backend
+      run: nix build .#game-night-backend
+
+    - name: Build Frontend
+      run: nix build .#game-night-frontend
+
+    - name: Build Docker Image
+      run: nix build .#game-night-docker
+
+    - name: Log in to the Container Registry
+      if: github.ref == 'refs/heads/main' && github.event_name == 'push'
+      uses: docker/login-action@v2
+      with:
+        registry: ${{ secrets.REGISTRY }}
+        username: ${{ secrets.DO_TOKEN }}
+        password: ${{ secrets.DO_TOKEN }}
+
+    - name: Push Docker Image
+      if: github.ref == 'refs/heads/main' && github.event_name == 'push'
+      run: |
+        # TODO - There's an action to extract the tags and label so we don't have to hardcode these
+        docker load < result
+        docker image ls | grep -i game-night
```

Comment:
Are these lines for CI log output or should they be removed?

Reply:
I need to clean this up. Take care of that TODO and adjust tagging strategy before we merge. Can get rid of some of that extra output also.